### ISSUE TITLE
occasional date parsing errors for PubSubChannelPointsMessage in ext/pubsub/models.py

### ISSUE THREAD

**Issue by charlesmadere** [TwitchIO/TwitchIO#227]
Sometimes, when constructing a new `PubSubChannelPointsMessage` instance, this line will cause an exception to be thrown:

```python
self.timestamp = datetime.datetime.strptime(
    redemption["redeemed_at"][0:25] + redemption["redeemed_at"][28:], "%Y-%m-%dT%H:%M:%S.%fZ"
)
```

It seems that the `Z` character doesn't always end up being a part of the string, and as such, the `strptime` call fails.

I wrote a really stupid fix for this, that seems to work for me:

Change the `self.timestamp` line within `PubSubChannelPointsMessage`'s `__init__` to this:
```python
self.timestamp = parseDate(redemption["redeemed_at"])
```

Then add this method somewhere within `ext/pubsub/models.py`:
```python
def parseDate(originalDateStr: str) -> datetime:
    dateStr = originalDateStr[0:25] + originalDateStr[28:]

    if dateStr[-1] != 'Z':
        dateStr = dateStr + 'Z'

    return datetime.datetime.strptime(dateStr, "%Y-%m-%dT%H:%M:%S.%fZ")
```

Interestingly, within `PubSubBitsBadgeMessage`'s `__init__`, there is another `striptime` call that looks identical to the one in `PubSubChannelPointsMessage`. This use however does not seem to have raised any exceptions for me. But regardless, I still use my above `parseDate()` method with it and it seems to be fine.

**Comment by EvieePy** [comment]
Thanks. This issue keeps popping up so I will take a look at it in the morning.

**Comment by charlesmadere** [comment]
this issue appears to be completely fixed as of release [`v2.1.0`](https://github.com/TwitchIO/TwitchIO/releases/tag/v2.1.0)
